### PR TITLE
Fix dynamic scoring

### PR DESCRIPTION
**Pull request by l-k-11235**
This PR is a fix for the dynamic scoring introduced is Open-NMT v.2 (see https://github.com/OpenNMT/OpenNMT-py/pull/2198) and not yet restored in v.3.

### CODE REVIEW

File: onmt/inputters/text_utils.py [hunk 1]
```diff
@@ -232,12 +232,21 @@ def textbatch_to_tensor(vocabs, batch):
     for i, ex in enumerate(batch):
         if isinstance(ex, bytes):
             ex = ex.decode("utf-8")
-        toks = ex.strip("\n").split()
+        if is_train:
+            toks = ex
+        else:
+            toks = ex.strip("\n").split()
```
Comment (francoishernandez) [inline onmt/inputters/text_utils.py:238]: This looks strange. What's the rationale?

File: onmt/translate/utils.py [hunk 2]
```diff
@@ -137,4 +137,11 @@ def translate(self, model, batch, gpu_rank, step, mode):
                     file.write("SOURCE: {}\n".format(sources[i]))
                     file.write("REF: {}\n".format(texts_ref[i]))
                     file.write("PRED: {}\n\n".format(preds[i]))
+        for layer in model.decoder.transformer_layers:
```
Comment (francoishernandez) [inline onmt/translate/utils.py:140]: Add a comment to explain why we reset the cache here.

File: onmt/utils/report_manager.py [hunk 3]
```diff
@@ -129,11 +129,11 @@ def _report_training(self, step, num_steps, learning_rate, patience,
...
-        self.maybe_log_tensorboard(report_stats,
-                                   "progress",
-                                   learning_rate,
-                                   patience,
-                                   step)
+        # self.maybe_log_tensorboard(report_stats,
+        #                            "progress",
+        #                            learning_rate,
+        #                            patience,
+        #                            step)
```
Comment (francoishernandez) [inline onmt/utils/report_manager.py:136]: ?

File: onmt/inputters/text_utils.py (hunk 1 above)
Comment (l-k-11235) [reply onmt/inputters/text_utils.py:238]: In training mode examples are already tokenized.

File: onmt/utils/report_manager.py (hunk 3 above)
Comment (l-k-11235) [reply onmt/utils/report_manager.py:136]: This provokes an inconsistent dispalying on tesorboard. Seems to be already done there
https://github.com/l-k-11235/OpenNMT-py/blob/7286ad98bf8df219ac009377e6cf04161e37d9d6/onmt/utils/report_manager.py#L156
https://github.com/l-k-11235/OpenNMT-py/blob/7286ad98bf8df219ac009377e6cf04161e37d9d6/onmt/utils/report_manager.py#L166

Comment (francoishernandez) [reply onmt/utils/report_manager.py:136]: Remove it then.
